features/topology-map/plugins/org.opennms.features.topology.plugins.topo.graphml/src/main/java/org/opennms/features/topology/plugins/topo/graphml/internal/Scripting.java

```diff
@@ -0,0 +1,178 @@
+    private final BinaryOperator<S> statusAccumulator;
+
+    public Scripting(final Path scriptPath,
+                     final ScriptEngineManager scriptEngineManager,
+                     final Supplier<S> statusDefault,
+                     final BinaryOperator<S> statusAccumulator) {
+        this.scriptPath = Objects.requireNonNull(scriptPath);
+        this.scriptEngineManager = Objects.requireNonNull(scriptEngineManager);
+        this.statusDefault = Objects.requireNonNull(statusDefault);
+        this.statusAccumulator = Objects.requireNonNull(statusAccumulator);
+    }
+
+    private List<StatusScript<S>> loadScripts() {
+        final List<StatusScript<S>> scripts = Lists.newArrayList();
+        try (final DirectoryStream<Path> stream = Files.newDirectoryStream(this.scriptPath)) {
+            for (final Path path : stream) {
+                final String extension = FilenameUtils.getExtension(path.toString());
+                final ScriptEngine scriptEngine = this.scriptEngineManager.getEngineByExtension(extension);
+                if (scriptEngine == null) {
+                    LOG.warn("No script engine found for extension '{}'", extension);
+                    continue;
+                }
+                LOG.debug("Found script: path={}, extension={}, engine={}", path, extension, scriptEngine);
+                try (final Stream<String> lines = Files.lines(path, Charset.defaultCharset())) {
+                    final String source = lines.collect(Collectors.joining("\n"));
+                    scripts.add(new StatusScript(scriptEngine, source));
+                }
+            }
+        } catch (final IOException e) {
+            LOG.error("Failed to walk template directory: " + this.scriptPath, e);
+            return Collections.emptyList();
+        }
+
+        return scripts;
+    }
+
+    private S computeStatus(final List<StatusScript<S>> scripts,
+                            final E element,
+                            final Function<E, SimpleBindings> refBindingMapper) {
+        return scripts.stream()
+                      .flatMap(script -> {
+                          final SimpleBindings bindings = refBindingMapper.apply(element);
+                          final StringWriter writer = new StringWriter();
+                          final ScriptContext context = new SimpleScriptContext();
+                          context.setWriter(writer);
+                          context.setBindings(bindings, ScriptContext.GLOBAL_SCOPE);
+                          try {
+                              LOG.debug("Executing script: {}", script);
+                              final S status = script.eval(context);
+                              if (status != null) {
+                                  return Stream.of(status);
+                              } else {
+                                  return Stream.empty();
+                              }
+                          } catch (final ScriptException e) {
```

Comment:
With the EdgeStatusProvider it was possible to see exceptions in the UI in the Info Panel Item. Is this still the case or do we swallow these exceptions now? I'd prefer to have them show up to make it easier for users to debug.